agrirouter-api-java-impl/src/main/java/com/dke/data/agrirouter/impl/messaging/mqtt/SendMessageServiceImpl.java

```diff
@@ -14,23 +16,35 @@
  * queue because connection handling should not be part of the API.
  */
 public class SendMessageServiceImpl extends MqttService
-    implements SendMessageService, MessageSender {
+    implements SendMessageService<Void>, MessageBodyCreator {
 
   public SendMessageServiceImpl(IMqttClient mqttClient) {
     super(mqttClient);
   }
 
-  public void send(SendMessageParameters parameters) {
-    parameters.validate();
+  /**
+   * Send message synchronous.
+   *
+   * @param sendMessageParameters Parameters to send the message.
+   */
```

Comment:
Is there a defined reason, why some function receive a function header for description (like this one) and some don't?(Like the next one).

Reply:
Not all the functions had comments in the former code. Every time I see one I will adapt the documentation, but not for each one I see. The Guideline should be, that all public functions & members should  have JavaDoc, private ones could have. So basically this will be an over the time improvement.